src/openrct2-ui/windows/RideList.cpp

```diff
@@ -172,872 +143,795 @@ static constexpr const rct_string_id page_names[] = {
                     {
-                        if (ride->satisfaction * 5 <= otherRide->satisfaction * 5)
-                            break;
-
-                        window_bubble_list_item(w, current_list_position);
+                        formatSecondary = STR_SATISFACTION_LABEL;
+                        ft.Add<uint16_t>(ridePtr->satisfaction * 5);
                     }
-                }
-                break;
-            case INFORMATION_TYPE_PROFIT:
-                while (--current_list_position >= 0)
-                {
-                    auto otherRide = get_ride(_rideList[current_list_position]);
-                    if (otherRide != nullptr)
+                    break;
+                case INFORMATION_TYPE_PROFIT:
+                    formatSecondary = 0;
+                    if (ridePtr->profit != MONEY64_UNDEFINED)
                     {
-                        if (ride->profit <= otherRide->profit)
-                            break;
-
-                        window_bubble_list_item(w, current_list_position);
+                        formatSecondary = STR_PROFIT_LABEL;
+                        ft.Add<money64>(ridePtr->profit);
                     }
-                }
-                break;
-            case INFORMATION_TYPE_TOTAL_CUSTOMERS:
-                while (--current_list_position >= 0)
-                {
-                    auto otherRide = get_ride(_rideList[current_list_position]);
-                    if (otherRide != nullptr)
+                    break;
+                case INFORMATION_TYPE_TOTAL_CUSTOMERS:
+                    formatSecondary = STR_RIDE_LIST_TOTAL_CUSTOMERS_LABEL;
+                    ft.Add<uint32_t>(ridePtr->total_customers);
+                    break;
+                case INFORMATION_TYPE_TOTAL_PROFIT:
+                    formatSecondary = 0;
+                    if (ridePtr->total_profit != MONEY64_UNDEFINED)
                     {
-                        if (ride->total_customers <= otherRide->total_customers)
-                            break;
-
-                        window_bubble_list_item(w, current_list_position);
+                        formatSecondary = STR_RIDE_LIST_TOTAL_PROFIT_LABEL;
+                        ft.Add<money64>(ridePtr->total_profit);
                     }
-                }
-                break;
-            case INFORMATION_TYPE_TOTAL_PROFIT:
-                while (--current_list_position >= 0)
+                    break;
+                case INFORMATION_TYPE_CUSTOMERS:
+                    formatSecondary = STR_RIDE_LIST_CUSTOMERS_PER_HOUR_LABEL;
+                    ft.Add<uint32_t>(ride_customers_per_hour(ridePtr));
+                    break;
+                case INFORMATION_TYPE_AGE:
                 {
-                    auto otherRide = get_ride(_rideList[current_list_position]);
-                    if (otherRide != nullptr)
+                    int16_t age = date_get_year(ridePtr->GetAge());
```

Comment:
```suggestion
                    const auto age = date_get_year(ridePtr->GetAge());
```
I'm not sure why the compiler did not warn, but its resulting int32_t.